lib/src/models/smart_auth_result.dart

```diff
@@ -2,7 +2,11 @@
 enum SmartAuthResultState {
   success,
   failure,
-  canceled,
+  canceled;
+
+  bool get isSuccess => this == SmartAuthResultState.success;
```

Comment:
It's better to make the `state` property private and expose `isCanceled` directly from the `SmartAuthResult` otherwise, it's getting confusing why we have to check `result.hasData`and `result.state.isSuccess`

Reply:
I guess it comes down to preferences. I prefer exposing the state, so user has option what to do with ui based on different state instead of exposing only selective state check.

Reply:
Basically, I took the structure from `AsyncSnapshot` because it's simple and people are used to it.

I'm okay with exposing the state property directly, but in that case we have to remove the `hasData/hasError` getters from `SmartAuthResult`, otherwise we would have two ways for doing the same thing, which might be confusing

Reply:
Alright, It was developed even before enhanced enum was not a thing. But i guess your package is already being used by so many people, so it will hassle for all those users to transition. I will revert this.